discovery_engine_core/core/src/lib.rs

```diff
@@ -27,7 +27,8 @@
 #![allow(
     clippy::missing_errors_doc,
     clippy::must_use_candidate,
-    clippy::module_name_repetitions
```

Comment:
This changes should be discussed with the team.
Not sure why allowing this `clippy::module_name_repetitions` sneaked in instead of discussing it with the others.

Reply:
Because we had this discussion before as far as I remember, it's also why I requested a re-review.

Reply:
Wait a moment something isn't right.

Reply:
What was added was `clippy::items_after_statements` not `clippy::module_name_repetitions`.

Reply:
In the `dart-api-dl` crate.

Reply:
`clippy::module_name_repetitions` was added recently after you asked me and I told you to discuss it with the team and this did not happen. You allow it on that struct but in general for project.
And for every change of the style is a team decision. For me is clear to have the struct declaration before the code. is also what often happen in the standard library code.

Reply:
But this is about items in a function body, which if more then 1-3 lines makes the function supper unreadable as it splits up the function signature and the function body, or different parts of the function body.

Like IMHO having items before statements is an anti-patterns as long as the items aren't very very compact. like a simple drop guard.

Reply:
Wrt. `clippy::module_name_repetitions` this seems to have accidentally slipt in I will start a discussion on slack.